File: Reverse Linked List - Leetcode 206/206. Reverse Linked List.java
class ListNode {
    int val;
    ListNode next;

    ListNode(){}    

    ListNode(int val){
        this.val = val;
    }

    ListNode(int val, ListNode next){
        this.val = val;
        this.next = next;
    }
}

class Solution {
    public ListNode reverseList(ListNode head){
        ListNode cur = head;
        ListNode prev = null;

        while (cur)
        {
            ListNode temp = cur.next;
            cur.next = prev;
            prev = cur;
            cur = temp;
        }

        return prev;
    }
}

File: Reverse Linked List - Leetcode 206/206. Reverse Linked List.py
class ListNode:
    def __init__(self,val=0,next=None):
        self.val = val
        self.next = next

def reverseList(head:ListNode) -> ListNode:
    cur = head;
    prev = null;

    while cur:
        temp = cur.next
        cur.next = prev
        prev = cur
        cur = temp
    
    return prev

